src/graphql/mod.rs

```diff
@@ -313,6 +347,90 @@ impl QueryClient {
     ) -> Result<Response<is_authorized_query::ResponseData>, APIError> {
         IsAuthorizedQuery::fetch(self, application, namespace, version).await
     }
+
+    #[wukong_telemetry(api_event = "deploy_livebook")]
+    pub async fn deploy_livebook(
+        &self,
+        application: &str,
+        namespace: &str,
+        version: &str,
+        name: &str,
+        port: i64,
+    ) -> Result<Response<deploy_livebook::ResponseData>, APIError> {
+        DeployLivebook::mutate(self, application, namespace, version, name, port).await
+    }
+
+    #[wukong_telemetry(api_event = "watch_livebook")]
+    pub async fn subscribe_watch_livebook(
+        &self,
+        variables: watch_livebook::Variables,
+    ) -> Result<
+        (
+            AsyncWebsocketClient<GraphQLClient, Message>,
+            SubscriptionStream<GraphQLClient, StreamingOperation<WatchLivebook>>,
+        ),
+        APIError,
+    > {
+        self.subscribe_graphql::<WatchLivebook>(variables).await
+    }
+
+    async fn subscribe_graphql<T: GraphQLQuery + Send + Sync + Unpin + 'static>(
+        &self,
+        variables: T::Variables,
+    ) -> Result<
+        (
+            AsyncWebsocketClient<GraphQLClient, Message>,
+            SubscriptionStream<GraphQLClient, StreamingOperation<T>>,
+        ),
+        APIError,
+    >
+    where
+        <T as GraphQLQuery>::Variables: Send + Sync + Unpin,
+        <T as GraphQLQuery>::ResponseData: std::fmt::Debug,
+    {
+        let bearer = format!("Bearer {}", self.access_token.clone().unwrap_or_default());
+
+        let mut request = "ws://localhost:4000/api/graphql-ws"
+            .into_client_request()
+            .unwrap();
+
+        request.headers_mut().append(
+            "Sec-WebSocket-Protocol",
+            HeaderValue::from_str("graphql-transport-ws").unwrap(),
+        );
+        request
+            .headers_mut()
+            .append("authorization", HeaderValue::from_str(&bearer).unwrap());
+
+        let (connection, _response) = async_tungstenite::tokio::connect_async(request)
+            .await
+            .unwrap();
```

Comment:
I got an error as below 

```
❯ cargo run -- application instances connect green@staging/mv-wukong-ci-mock-green-69b78dfd88-7fj9j
    Finished dev [unoptimized + debuginfo] target(s) in 0.69s
     Running `target/debug/wukong application instances connect 'green@staging/mv-wukong-ci-mock-green-69b78dfd88-7fj9j'`
Checking your permission to connect to the remote instance...✅
⠙ Preparing your remote instance...                                                                                                                                               thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Http(Response { status: 400, version: HTTP/1.1, headers: {"cache-control": "max-age=0, private, must-revalidate", "content-length": "53", "content-type": "application/json; charset=utf-8", "date": "Wed, 31 May 2023 09:31:39 GMT", "server": "Cowboy", "x-request-id": "F2QwguU81p7pNS8AAAQS"}, body: Some([123, 34, 101, 114, 114, 111, 114, 115, 34, 58, 91, 123, 34, 109, 101, 115, 115, 97, 103, 101, 34, 58, 34, 78, 111, 32, 113, 117, 101, 114, 121, 32, 100, 111, 99, 117, 109, 101, 110, 116, 32, 115, 117, 112, 112, 108, 105, 101, 100, 34, 125, 93, 125]) })', /Users/tuanco/code-workspaces/rust/src/github.com/mindvalley/wukong-cli/src/graphql/mod.rs:407:14
```

In production build, how are we handling this error response ?

Reply:
let me check on this